src/modules/logger/logger.cpp

```diff
@@ -518,6 +520,29 @@ void Logger::run()
 {
 	PX4_INFO("logger started (mode=%s)", configured_backend_mode());
 
+	bool boot_logging_bat_only = false;
+	bool disable_boot_logging = false;
+
+	if (_boot_bat_only != PARAM_INVALID) {
+		param_get(_boot_bat_only, &boot_logging_bat_only);
```

Comment:
`boot_logging_bat_only` needs to be `int32_t`

@dagar why did you change that in https://github.com/PX4/Firmware/commit/060463e8a7436701394f0ebe2d4ce568c7780289#diff-49dacc90dcc99405772dbb3540d7665fL485? It would have been caught.